cmd/examplar/README.md

```diff
@@ -0,0 +1,133 @@
+
+Every original line in the input SQL file is executed and expected to not return any error.
+
+Those test files have no dependencies on Examplar or the SQL file that has been used to generate the test.
+
+### Annotations
+
+Annotations starts with `---` (the SQL comment `--` and an additional `-`) and can
+be followed by a keyword specified in the list below followed by a `:` or special symbols to pass data to set expectations.
+
+- `setup:`
+
+  - all lines up to the next annotation are to be considered as individual statements for the setup block.
+
+- `teardown:`
+  - all lines up to the next annotation are to be considered as individual statements for the teardown block.
+
+:bulb: `setup` and `teardown` blocks will generate code being ran around **each indidual** `test` block.
+They are optional and can be declared in no particular order, but there can only be one of each.
+
+- `test: [TEST NAME]`
+  - all lines up to the next annotation are to be considered as individual statements for this test block. Each line can be followed by the special expectation annotation.
+
+:bulb: Each `test` block will generate an individual `t.Run("[TEST NAME]", ...)` function. At least one test block must be present.
+
+- `` `[JSON]` ``
+
+  - the previous line evaluation result will be compared to the given `[JSON]` value.
+  - Invalid JSON won't yield an error ar generate time, but the generated test will always fail at runtime.
+
+- ` ``` `
+  - all the following lines until another triple backtick annoattion is found are to be considered as a multiline JSON data.
+  - The line preceding the the opening triple backticks will be evaluation result will be compared to the given JSON data.
+  - Indentation will be preserved in the generating test file for readablilty. Similarly, invalid JSON will only yield an error when the resulting test is evaluated.
+
+## Goals and non-goals
+
+Examplar objective is to provide a clear and simple way to write example of Genji SQL code and its expected results. It has to be easy enough for anyone to edit or write an example SQL file without having to read the present documentation. If this objective succeeds, it opens the path Examplar SQL files being used by users and contributors to showcase a bug or a feature request in a Github issue.
+
+By being totally independent from Genji itself for the parsing part, it frees itself from needing to be updated when something changes under the hood and allows to keep the code as merely parsing textual data. In other words, if anything changes in Genji on how to execute queries, only the template needs to be updated.
+
+## Limitations
+
+A breaking change in the API has only
+For now, let's observe how useful Examplar can and what we can make out of it.
+Then we can then see if it's worth addressing the following limitations:
+
+- (1) Multilines SQL statements are not yet supported.
```

Comment:
TODO: swallow all lines until we get an assertion and pass them to `db.Query`, only the last one will return its results.  This may not be applicable if examplar is extracted, but with Genji it works perfectly.